sentry-kotlin-extensions/build.gradle.kts

```diff
@@ -0,0 +1,66 @@
+import net.ltgt.gradle.errorprone.errorprone
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+    `java-library`
+    kotlin("jvm")
+    jacoco
+    id(Config.QualityPlugins.errorProne)
+    id(Config.QualityPlugins.gradleVersions)
+    id(Config.QualityPlugins.detektPlugin)
+}
+
+configure<JavaPluginConvention> {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
+
+tasks.withType<KotlinCompile>().configureEach {
+    kotlinOptions.jvmTarget = JavaVersion.VERSION_1_8.toString()
+}
+
+dependencies {
+    api(project(":sentry"))
+    implementation(Config.Libs.coroutinesCore)
```

Comment:
could this be `compileOnly`? ideally `sentry-kotlin-extensions` would offer method extensions as well in the future, making it easier/friendlier working with Kotlin and the Unified API, but right now it depends on coroutines and people might not like it, another option would be to rename it, to be something more coroutines specific, @bruno-garcia opinions?

Reply:
Are there valid use cases where folks can't have coroutines in their project? Other than perhaps small edge cases?
Personally I rather we have a single package with all kotlin related bits like kotlin extensions and the coroutine support and make it a "all or nothing" thing. Breaking into 2 packages will add to the complexity in terms of maintenance for us, publshing more packages to central, documenting extra packages.

Maybe I'll rephrase the question: If I"m creating a Kotlin project today, is there a reason I want it not to have?

Reply:
the problem is on Android, it adds 1.9 MB as a dependency, and coroutines got traction also on Android apps but still not adopted by everyone, and dependency's size is one of the problems.
I see though Android apps wanting to use Sentry extensions, we can live with that, but it's a trade-off that I likely would not have as an Android dev

Reply:
Would it be fair to say that given extension method (ktx) is not really a requirement, it's bundled in the `kotlin-extensions` package which includes coroutines. For users who have not adopted coroutines, they are not blocked, they continue to have the support we have today, that is: no extension methods?
Not necessarily the best for those users but best balance between maintenance burden to us and future thinking (Assuming that new projects with kotlin will use coroutines)?

Reply:
lets publish as it is, once we're done with method extensions for Sentry, we evaluate this again, most Android people are starting with coroutines or willing to migrate it whenever possible (eg new features), so I agree that a balance here plays a bigger role, thanks